tests/Validation/RegExPatternValidationTest.php

```diff
@@ -0,0 +1,213 @@
+<?php
+
+declare( strict_types = 1 );
+
+namespace WMDE\Fundraising\Content\Tests\Validation;
+
+use PHPUnit\Framework\TestCase;
+
+class RegExPatternValidationTest extends TestCase {
+
+	/**
+	 * @dataProvider validStreetNameProvider
+	 */
+	public function testValidStreetNames( string $streetName ): void {
+		$addressValidationObject_DE =  $this->loadAddressValidationPatternsFromFile( 'de_DE/data/validation.json' );
+		$pattern = $addressValidationObject_DE['street'];
+
+		$this->assertMatchesRegularExpression( '/'.$pattern.'/', $streetName);
```

Comment:
```suggestion
		$this->assertMatchesRegularExpression( "/$pattern/", $streetName);
```
I was briefly confused by the dots, thought they were part of the regex, then realized that they're PHP. 

I know we try to use single quotes in most of our code, but in this case I think double quotes and interpolation makes the code more readable. It's a tradeoff between consistency and readability, and if you like the consistency, please add some space around the dots instead.

BTW, do you know *why* we mostly use single quotes? AFAIK are two reasons:

1) to avoid accidental string interpolation 
2) as a habit from ancient versions of PHP where using single quotes made code run faster because it skipped interpolation. This is no longer true for PHP >= 7.4, see https://www.phpbench.com/

Reply:
forgot that something like easy string interpolation existed. that naive part of my programmer brain died some time ago TT